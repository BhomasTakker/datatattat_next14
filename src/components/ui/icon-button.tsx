import { StyleSheet } from "@/types/css";
import { ButtonHTMLAttributes, FC } from "react";
import styles from "./icon-button.module.scss";

type IconButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
	icon: FC;
	onClick: () => void;
	classes?: StyleSheet;
	tooltip?: string;
};

export const IconButton = ({
	icon,
	onClick,
	tooltip,
	type = "button",
}: IconButtonProps) => {
	const Icon = icon;

	return (
		<button className={styles.icon} onClick={onClick} type={type}>
			<Icon />
			{tooltip && <p className={styles.tooltip}>{tooltip}</p>}
		</button>
	);
};
