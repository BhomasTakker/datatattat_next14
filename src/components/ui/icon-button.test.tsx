import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { IconButton } from "./icon-button";

// Mock icon component
const MockIcon: React.FC = () => <svg data-testid="mock-icon" />;

describe("IconButton", () => {
	it("renders the icon", () => {
		const { getByTestId } = render(
			<IconButton icon={MockIcon} onClick={() => {}} />
		);
		expect(getByTestId("mock-icon")).toBeInTheDocument();
	});

	it("calls onClick when clicked", () => {
		const handleClick = jest.fn();
		const { getByRole } = render(
			<IconButton icon={MockIcon} onClick={handleClick} />
		);
		fireEvent.click(getByRole("button"));
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	it("sets the button type to 'button' by default", () => {
		const { getByRole } = render(
			<IconButton icon={MockIcon} onClick={() => {}} />
		);
		expect(getByRole("button")).toHaveAttribute("type", "button");
	});

	it("accepts and sets a custom button type", () => {
		const { getByRole } = render(
			<IconButton icon={MockIcon} onClick={() => {}} type="submit" />
		);
		expect(getByRole("button")).toHaveAttribute("type", "submit");
	});

	it("applies the correct className from styles", () => {
		const { getByRole } = render(
			<IconButton icon={MockIcon} onClick={() => {}} />
		);
		expect(getByRole("button").className).toContain("icon");
	});

	describe("Snaps", () => {
		it("renders IconButton unchanged", async () => {
			const { container } = render(
				<IconButton icon={MockIcon} onClick={() => {}} />
			);
			expect(container).toMatchSnapshot();
		});
	});
});
