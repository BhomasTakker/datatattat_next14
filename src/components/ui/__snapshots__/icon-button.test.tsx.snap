// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`IconButton Snaps renders IconButton unchanged 1`] = `
<div>
  <button
    class="icon"
    type="button"
  >
    <svg
      data-testid="mock-icon"
    />
  </button>
</div>
`;
